pkg/controller/clusterserviceversion/clusterserviceversion_controller.go

```diff
@@ -196,5 +278,98 @@ func (r *ReconcileClusterServiceVersion) Reconcile(request reconcile.Request) (r
 	}
 
 	reqLogger.Info("reconcilation complete")
-	return reconcile.Result{}, nil
+	return reconcile.Result{RequeueAfter: time.Minute * 1}, nil
+}
+
+// deleteExternalResources searches for the MeterDefinition created by the CSV, if it's found delete it
+func (r *ReconcileClusterServiceVersion) deleteExternalResources(CSV *olmv1alpha1.ClusterServiceVersion) error {
+	reqLogger := log.WithValues("Request.Name", CSV.GetName(), "Request.Namespace", CSV.GetNamespace())
+	reqLogger.Info("deleting csv")
+	var err error
+
+	meterDefinitionList := &marketplacev1alpha1.MeterDefinitionList{}
+	err = r.client.List(context.TODO(), meterDefinitionList, client.InNamespace(CSV.GetNamespace()))
+	if err != nil {
+		return err
+	}
+
+	// search specifically for the Meter Definition created by the CSV
+	for _, meterDefinition := range meterDefinitionList.Items {
+		ann := meterDefinition.GetAnnotations()
+		if _, ok := ann[utils.CSV_ANNOTATION_NAME]; ok {
+			// CSV specific MeterDefinition found; delete it
+			err = r.client.Delete(context.TODO(), &meterDefinition, client.PropagationPolicy(metav1.DeletePropagationForeground))
+			if err != nil {
+				return err
+			}
+			reqLogger.Info("found and deleted MeterDefinition")
+			return nil
+		}
+	}
+
+	reqLogger.Info("no meterdefinition found")
+	return nil
+}
+
+// checkAnnotations checks the Annotations for the rhm CSV
+// If the CSV is new, we tag it and create a MeterDefinition
+// If the CSV is old, we check if the existing MeterDefinition matches the CSV (json) MeterDefinition
+func (r *ReconcileClusterServiceVersion) checkAnnotations(CSV *olmv1alpha1.ClusterServiceVersion, annotations map[string]string) error {
```

Comment:
LIke breaking this out. We should return a reconcile.Result{} and a bool to requeue as well. We need to requeue after an update because it can cause later calls to fail if we don't. Also, name is a bit suspect (I'm also terrible at names). Maybe reconcileMeterDefAnnotation may better describe what this function is doing.
```suggestion
func (r *ReconcileClusterServiceVersion) reconcileMeterDefAnnoations(CSV *olmv1alpha1.ClusterServiceVersion, annotations map[string]string) (reconcile.Result, bool, error) {
```